### PR TITLE
feat: Compute graph and CM sheet extension

### PR DESCRIPTION
**Pull request by richburdon**
- [x] Restructure to isolate HF defs.
- [x] ComputeGraphRegistry root class.
- [x] Functionmanager abstraction.
- [x] CM extension for graph. 
- [x] Storybooks
- [x] Async import dep from plugin to HF.
- [x] Factor out HF dependency from SheetPlugin.
- [ ] Browser and non-browser tests.
- [ ] Factor out Space dependency from SheetModel and generalize compute dependency graph.

<br />

Screenshot 2024-09-28 at 6 32 57 PM

<br />

Screenshot 2024-09-29 at 12 49 08 PM

<br />

https://github.com/user-attachments/assets/a529b05e-9d75-400a-a205-471053fecb18

### CODE REVIEW

File: packages/plugins/plugin-sheet/src/graph/compute-node.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  // TODO(burdon): Remove?
+  get graph() {
+    return this._graph;
+  }
+
+  get hf() {
+    return this._graph.hf;
+  }
+
+  getValue(cell: CellAddress): CellScalarValue {
```
Comment (dmaretskyi) [inline packages/plugins/plugin-sheet/src/graph/compute-node.ts:38]: Is there a way to support other addressing formats for non-sheet objects? For example acessing table data

Comment (wittjosiah) [reply packages/plugins/plugin-sheet/src/graph/compute-node.ts:38]: you just need to be able to map to a cell address

File: packages/plugins/plugin-sheet/src/model/sheet-model.browser.test.ts [hunk 2]
```diff
@@ -4,43 +4,50 @@
...
+    const sheet = createSheet({ rows: 5, columns: 5 });
+    const model = new SheetModel(graph, sheet);
+    await model.open();
+    return { graph, model };
   };
 
   test('create', async () => {
-    const model = await createModel();
+    const { model } = await createModel();
+    console.log(model);
```
Comment (wittjosiah) [inline packages/plugins/plugin-sheet/src/model/sheet-model.browser.test.ts:35]: 👀

File: packages/plugins/plugin-table/src/components/ObjectTable/ObjectTable.tsx [hunk 3]
```diff
@@ -29,9 +29,15 @@ export type ObjectTableProps = Pick<TableProps<any>, 'stickyHeader' | 'role'> &
 export const ObjectTable = ({ table, role, stickyHeader }: ObjectTableProps) => {
   const space = getSpace(table);
   const [showSettings, setShowSettings] = useState(false);
-
   useEffect(() => setShowSettings(!table.schema), [table.schema]);
 
+  const [schemas, setSchemas] = useState<DynamicSchema[]>([]);
+  useEffect(() => {
+    if (space) {
+      void space.db.schema.list().then(setSchemas).catch();
```
Comment (wittjosiah) [inline packages/plugins/plugin-table/src/components/ObjectTable/ObjectTable.tsx:37]: todo don't don't void promise in useEffect

File: packages/sdk/client/src/echo/index.ts [hunk 4]
```diff
@@ -12,8 +12,9 @@ export {
   getTypename,
   Expando,
   type EchoReactiveObject,
-  type ReactiveObject,
   type ObjectMeta,
+  type ReactiveObject,
+  type S, // TODO(burdon): Export as Schema?
```
Comment (wittjosiah) [inline packages/sdk/client/src/echo/index.ts:17]: wasn't it suggested that this shouldn't be reexported?

File: packages/plugins/plugin-sheet/src/SheetPlugin.tsx [hunk 5]
```diff
@@ -6,50 +6,43 @@ import { GridNine, type IconProps } from '@phosphor-icons/react';
...
-
+      invariant(client);
+      let remoteFunctionUrl: string | undefined;
       if (client.config.values.runtime?.services?.edge?.url) {
         const url = new URL('/functions', client.config.values.runtime?.services?.edge?.url);
         url.protocol = 'https';
         remoteFunctionUrl = url.toString();
       }
+
+      // Async import removes direct dependency on hyperformula.
```
[the remark this answers is not in the document]
Comment (dmaretskyi) [reply packages/plugins/plugin-sheet/src/SheetPlugin.tsx:38]: everything in this file will be loaded during composer startup. Hyperformula is not small, so we shouldn't block startup on it, especcially if the user is not opening a spreadsheet

File: packages/plugins/plugin-table/src/components/ObjectTable/ObjectTable.tsx (hunk 3 above)
Comment (richburdon) [reply packages/plugins/plugin-table/src/components/ObjectTable/ObjectTable.tsx:37]: why?

File: packages/sdk/client/src/echo/index.ts (hunk 4 above)
Comment (richburdon) [reply packages/sdk/client/src/echo/index.ts:17]: we're currently re-exporting it from echo-schema.
do you think we shoudldn't do that?

File: packages/plugins/plugin-sheet/src/graph/compute-node.ts (hunk 1 above)
Comment (richburdon) [reply packages/plugins/plugin-sheet/src/graph/compute-node.ts:38]: no, for tables, etc. we'd just use ECHO queries, no?

File: packages/plugins/experimental/plugin-debug/src/components/ObjectCreator.tsx
[the remark this answers is not in the document]
Comment (richburdon) [reply packages/plugins/experimental/plugin-debug/src/components/ObjectCreator.tsx:1]: i would but seemed too complex to get working.

**Pull request merged**